### PR TITLE
Add `String` methods.

### PR DESCRIPTION
**Pull request by maks**
Adds almost all of the remaining `String` methods.

### CODE REVIEW

**Comment by maks** [comment]
@ethanblake4 sorry I've dropped the ball on this a bit, I'll try to get it finished off soon.

**Comment by ethanblake4** [comment]
No worries. Looking forward to it :)

**Comment by maks** [comment]
@ethanblake4 this is ready for you to have a look at when you get a chance. I've tried to have at least one test per each method, hopefully the coverage is ok. I'll be happy to come back with another PR for the remaining methods once the dependent classes are available.

File: test/dart_eval_test.dart [hunk 1]
```diff
@@ -876,6 +876,301 @@ void main() {
...
+              const cat = "Fluffy";
+              return cat.toString();
+            }
+          ''',
+        }
+      });
+      expect((exec.executeLib('package:example/main.dart', 'test') as $String).$value, 'Fluffy');
+    });
+
+// Doesn't work right now, seems to ignore that KnownMethodArg has nullable set to true and get this error:
```
Comment (maks) [inline test/dart_eval_test.dart:1090]: @ethanblake4 I'm not sure if I'm doing something wrong here or if it is a bug in the nullable parameter handling?

**Comment by ethanblake4** [comment]
This looks awesome! For replaceRange I've just pushed some changes to support nullability in argument lists. The syntax is a little different, instead of just putting `true` for nullable in the KnownMethodArg (which still doesn't do anything) you should specify the arg as (for example): 

```dart
EvalTypes.intType.copyWith(nullable: true)
```

I'm going to remove the nullable parameter from KnownMethodArg althogether, but I'll wait until we get this PR merged.

**Comment by maks** [comment]
Awesome, thanks @ethanblake4 !
I'll update the PR using that shortly.

**Comment by maks** [comment]
@ethanblake4 I've pushed up the change to make use of the new way to handle nullable parameters,

**Comment by maks** [comment]
@ethanblake4 sorry one more thing: I've noticed that all the tests for String methods have made `dart_eval_test.dart` quite huge, would you like me to split the String method tests out into a separate file?

**Comment by ethanblake4** [comment]
I've been noticing that too, if you'd like to do that that'd be great!

**Comment by maks** [comment]
@ethanblake4 done 👍🏻

**Comment by ethanblake4** [comment]
Thanks @maks !

**Comment by ethanblake4** [comment]
Released v0.5.4 with these changes

**Pull request merged**